src/agoradatatools/etl/utils.py

```diff
@@ -142,6 +157,20 @@ def nest_fields(
         .rename(columns={0: new_column})
     )
 
+    if nested_field_is_list:
+        return nested
+
+    # nested_field_is_list == False
+    lengths = nested[new_column].apply(len)
+    if all(lengths == 1):
+        nested[new_column] = nested[new_column].apply(lambda row: row[0])
+        return nested
+    else:
+        raise ValueError(
+            "nested_field_is_list cannot be False when there are multiple rows to nest per "
+            + grouping
+        )
```

Comment:
@BWMac or @thomasyu888 is ValueError a good exception to raise here or should it be doing something else?

Reply:
I think `ValueError` is appropriate.